# First Assignment

Each correct answer is worth 1 point. Each wrong answer is worth 0 points.

**Deadline: Tuesday 31 May 2022, 18:00 BST**. Append your name to the notebook and submit it to me via email (massimiliano.izzo@conted.ox.ac.uk) or via Slack.

### Question A

Of the following statements, which are true and which are false?

  1. Anomaly detection is an example of supervised learning - TRUE
  2. Regression is a type of supervised learning - TRUE
  3. Self-supervised learning and semi-supervised learning are synonyms FALSE
  4. Regularisation is a technique used to tune the hyperparameters of a machine learning model FALSE
  5. Most of the machine learning algorithms require the data to be scaled. TRUE -> (so as to speedup the model)

*You can edit this cell to fill in you answers*

### Question B

Consider the following snippet of code:

In [22]:
from sklearn.datasets import make_circles
from sklearn.model_selection import train_test_split

X, y = make_circles(n_samples=1_000, factor=0.3, noise=0.05, random_state=0)
X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, random_state=0)

In [23]:
np.isnan(X_train).any() # check for NaN , missing record

False

In [24]:
X_train.shape # here we have two feature because no of columns are two

(750, 2)

In [28]:
set(y_train) # it is having only two class i.e. 0 and 1

{0, 1}

Of the following statements, which are true and which are false?

  1. The test set contains 33% of the samples of the training set FALSE -> (by default its 25%)
  2. None of the records in the training set has missing values TRUE
  3. `make_circles` generates a dataset TRUE -> its generating x and y as data
  4. The resulting dataset contains two features TRUE
  5. The resulting dataset can be used for multi-class classification FALSE -> (only two class)

*You can edit this cell to fill in you answers*

### Question C

Consider the code below:

In [12]:
import numpy as np
from sklearn.linear_model import SGDRegressor
from sklearn.datasets import make_regression
from sklearn.metrics import mean_squared_error

X, y = make_regression(n_samples=200, n_features=2, noise=0.2)
sgd_reg = SGDRegressor(
    loss="squared_error",
    max_iter=1000,
)
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    train_size=0.8,
    random_state=0
)
sgd_reg.fit(X_train, y_train)

SGDRegressor()

In [13]:
X_train.shape

(160, 2)

Of the following statements, which are true and which are false?

  1. The regressor `sgd_reg` has learned three parameters FALSE
  2. The regressor is trained on 180 data instances FALSE ->(trained on only 160 data)
  3. `sgd_reg` is trained to fit a logistic regressor FALSE ->(linear regression)
  4. The initial learning rate is a hyperparameter of `sgd_reg` TRUE
  5. `sgd_reg` is a linear model TRUE ->(its a linear Model)

*You can edit this cell to fill in you answers*